src/EditorFeatures/Core/ContentTypeLanguageMetadata.cs

```diff
@@ -2,28 +2,14 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System.Collections.Generic;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Editor
-{
-    internal class ContentTypeLanguageMetadata : LanguageMetadata
-    {
-        public string DefaultContentType { get; }
-
-        public ContentTypeLanguageMetadata(IDictionary<string, object> data)
-            : base(data)
-        {
-            this.DefaultContentType = (string)data.GetValueOrDefault("DefaultContentType");
-        }
+namespace Microsoft.CodeAnalysis.Editor;
 
-        public ContentTypeLanguageMetadata(string defaultContentType, string language)
-            : base(language)
-        {
-            this.DefaultContentType = defaultContentType;
-        }
-    }
+internal sealed class ContentTypeLanguageMetadata(IDictionary<string, object> data) : ILanguageMetadata
```

Comment:
intentional that it doesn't derive from LanguageMetadata?

Reply:
Yes.

Reply:
I think we should avoid inheriting metadata classes that correspond to unrelated Export attributes like in this case.
We still have the interface impl to be able to share code that uses the Language property.